### ISSUE TITLE
Rework README

### ISSUE THREAD

**Issue by StephaneCapponi** [socon-dev/socon#24]
The concept of Socon does not seems to be fully understand when people read the README. The example is too fictional and people can't relate to real word issue. 

The concept of the framework and what issue it solved needs to be clearer.

**Comment by JRbemt** [comment]
I agree, I think this project has much more potential than is currently communicated through the README. Maybe rephrasing it as a microservice architecture framework, and  focussing on deploying/chaining individual commands would be a an idea!? 
 (e.g. retrieving data from different databases [projectconfigs] and manipulate that with a basecommand and maybe finally plot some results to get data insights) 

Additionally I think some example projects are also needed.

**Comment by StephaneCapponi** [comment]
Yes you are right. I like the idea of microservice architecture. 

This is a good example:
`(e.g. retrieving data from different databases [projectconfigs] and manipulate that with a basecommand and maybe finally plot some results to get data insights)`

Let me give you some context on why I have created this repos and the README that way. I'm a Test and Validation engineer and I run a continuous integration system using Bamboo/Jenkins on mulitple projects. Each project is a different embedded platform. For each platform I have to build and flash similar binary with different configuration to test the stack develop by people. We are currently using mulitple shell scripts to run the CI and it's a mess. Genericity is bad and we have mulitple scripts for different projects that are copy/paste with just couple of lines that are different. 

In my spare time I have decided to create that framework to make my life easier. At first I wanted the framework to be embedded oriented but then I told myself that it can be used for anything and by anyone if I make it generic enough.

Today, I really believe that this framework can be usefull for a lot of people but my ideas and the README are maybe to oriented to the problem I had. Having a different view on the project is really important for me and I'm glad you like the project and you want to contribute. 

The goal is to widely share that framework. But the README is the first thing that people see and it's the priority. If you are ok, we need to work on that together and come with something great ?

**Comment by JRbemt** [comment]
Hey @StephaneCapponi, sounds good! I'm a bit swamped with work at the moment, but I will make some time. I think the main thing the README at the moment is missing is purpose. Interesting to hear though how you designed it for CI for different embedded platforms!! Did you ever use the framework in your work already?

**Comment by StephaneCapponi** [comment]
Hi, @JRbemt no problem I'm not in a hurry :) I will try to work on something on my side and find a purpose as you said. And yes I'm trying to find time at work to implement the framework!

**Comment by StephaneCapponi** [comment]
Hi @JRbemt, I have started the implementation of the framework at work and I'm creating new features and reporting bug in documentation or in the code. I saw that you have self-assigned this ticket to you. I really like the approach of saying that Socon is a Microservice framework. Something like:

Socon is a microservice architecture framework the helps you focussing on
deploying/chaining individual commands between your different projects. Socon's create
a generic Django like project structure. It's a structure that contains all the element
for you to get started.

I think it would be best that we remove the "Why" section and focus more on mulitple but simple example. Also I was wondering that if we call socon a microservice framework that all "project" should be "service". WDYT ?
